Mathlib/Analysis/Normed/Order/UpperLower.lean

```diff
@@ -148,10 +148,9 @@ lemma dist_le_dist_of_le_pi (ha : a₂ ≤ a₁) (h₁ : a₁ ≤ b₁) (hb : b
 theorem IsUpperSet.exists_subset_ball (hs : IsUpperSet s) (hx : x ∈ closure s) (hδ : 0 < δ) :
     ∃ y, closedBall y (δ / 4) ⊆ closedBall x δ ∧ closedBall y (δ / 4) ⊆ interior s := by
   refine ⟨x + const _ (3 / 4 * δ), closedBall_subset_closedBall' ?_, ?_⟩
-  · rw [dist_self_add_left]
-    refine (add_le_add_left (pi_norm_const_le <| 3 / 4 * δ) _).trans_eq ?_
-    simp only [norm_mul, norm_div, Real.norm_eq_abs]
-    simp only [zero_lt_three, abs_of_pos, zero_lt_four, abs_of_pos hδ]
+  · grw [dist_self_add_left, ← const_def, pi_norm_const_le <| 3 / 4 * δ]
+    apply le_of_eq
+    simp only [norm_mul, norm_div, Real.norm_eq_abs, abs_of_pos hδ]
```

Comment:
Is a mere `simp` appreciably slower? (Local profiling/eyeballing suffices for me.) If not, can you use that, please?